### DISCUSSION TITLE
No pdftk formula/cask?

### DISCUSSION THREAD

**Discussion by alexreg** [Homebrew/discussions#1552] [Casks]
Is there any reason there is no cask for [pdftk](https://www.pdflabs.com/tools/pdftk-server/)? Or even a formula, since source seems to be available.

**Comment by tjluoma** [comment]
I believe that `pdftk` is 32-bit, meaning that it will not run on anything newer than Mojave.

I'm not sure if that's why there's no cask for it… the program hasn't been updated in years, so it would be easy enough to point to the .pkg for those who do want to install/run it.

**Reply by alexreg** [reply]
The link on the website is to the old 32-bit version, but there a 64-bit version has been released, the latest one being:

https://www.pdflabs.com/tools/pdftk-the-pdf-toolkit/pdftk_server-2.02-mac_osx-10.11-setup.pkg

This runs fine on my macOS 11 system.

The source is also linked on the website... I haven't tried building it (on a 64-bit machine), but I imagine it should work.

**Comment by gromgit** [comment]
The most common reason for a specific Homebrew formula/cask _not_ existing is "no one wants it enough to put in the formula/cask-writing effort".

That said, there's already an alternative formula:
```bash
$ brew search pdftk
==> Formulae
pdftk-java

$ brew info pdftk-java
pdftk-java: stable 3.2.2 (bottled), HEAD
Port of pdftk in java
https://gitlab.com/pdftk-java/pdftk
Not installed
From: https://github.com/Homebrew/linuxbrew-core/blob/HEAD/Formula/pdftk-java.rb
License: GPL-2.0-or-later
==> Dependencies
Build: gradle ✘
Required: openjdk@8 ✘
==> Options
--HEAD
	Install HEAD version
==> Analytics
install: 5 (30 days), 11 (90 days), 85 (365 days)
install-on-request: 5 (30 days), 11 (90 days), 85 (365 days)
build-error: 0 (30 days)
```

**Reply by gromgit** [reply]
And specifically for `pdftk`, there seemed to have been a deal-breaking issue regarding unexpected permissions changes on the part of the `pdftk` installer: https://github.com/Homebrew/homebrew-cask/issues/7707

You might want to check if this is still being done by the installer you successfully run.

**Reply by alexreg** [reply]
> The most common reason for a specific Homebrew formula/cask _not_ existing is "no one wants it enough to put in the formula/cask-writing effort".

True, although this is fairly popular software, so that would slightly surprise me.

The permissions issue still exists (it messes up /usr/local/bin rather than /usr/local, since the latter falls under SIP now) — though I suspect the best solution is to create a formula and build from source anyway. Maybe I can look into that.
